### PR TITLE
Feature/client server verification

### PR DESCRIPTION
**Pull request by HugoZink**
Server kan nu reageren op challenges, om te bewijzen dat hij ook de private key bezit.

### CODE REVIEW

File: Triangle Streaming Server/ReceiveStream.cs [hunk 1]
```diff
@@ -77,12 +72,39 @@ protected override void OnMessage(MessageEventArgs e)
...
 			}
-			else if(e.Data.Equals("LIST"))
+			else if(e.Data.StartsWith("CHALLENGE: "))
 			{
-				//Send a list of streams to the client.
-				var streams = JsonConvert.SerializeObject(StreamQueueManager.GetInstance().Streams.Values.ToList());
+				string challengeString = e.Data.Replace("CHALLENGE: ", "");
+
+				//Convert base64 string back to encrypted bytes
+				var challengeBytes = Convert.FromBase64String(challengeString);
```
Comment (markieo1) [inline Triangle Streaming Server/ReceiveStream.cs]: To string

File: Triangle Streaming Server/ReceiveStream.cs [hunk 2]
```diff
@@ -77,12 +72,39 @@ protected override void OnMessage(MessageEventArgs e)
...
 
-				this.Send(streams.ToString());
+				//Decrypt bytes into string
+				IAsymmetricBlockCipher eng = new Pkcs1Encoding(new RsaEngine());
+				eng.Init(false, _keyPair.Private);
+				var responseString = Encoding.UTF8.GetString(eng.ProcessBlock(challengeBytes, 0, challengeBytes.Length));
```
Comment (markieo1) [inline Triangle Streaming Server/ReceiveStream.cs]: to string

File: Triangle Streaming Server/ReceiveStream.cs [hunk 3]
```diff
@@ -77,12 +72,39 @@ protected override void OnMessage(MessageEventArgs e)
...
+			{
+				switch(e.Data)
+				{
+					case "PUBKEY":
+						//Send public key to client
+						Send(_pubKey);
+						break;
+					case "LIST":
+						//Send a list of streams to the client.
+						var streams = JsonConvert.SerializeObject(StreamQueueManager.GetInstance().Streams.Values.ToList());
```
Comment (markieo1) [inline Triangle Streaming Server/ReceiveStream.cs]: to correct type, and is a null ref possible here?

File: Triangle Streaming Server/Triangle Streaming Server.csproj [hunk 4]
```diff
@@ -61,7 +61,9 @@
     <Compile Include="WebSocketManager.cs" />
   </ItemGroup>
   <ItemGroup>
-    <None Include="App.config" />
+    <None Include="App.config">
+      <SubType>Designer</SubType>
```
Comment (markieo1) [inline Triangle Streaming Server/Triangle Streaming Server.csproj]: Why is this added?

Comment (HugoZink) [inline Triangle Streaming Server/Triangle Streaming Server.csproj]: Must have been automatic. Will remove.

File: Triangle Streaming Server/ReceiveStream.cs (hunk 3 above)
Comment (HugoZink) [inline Triangle Streaming Server/ReceiveStream.cs]: Null reference should not be possible, the Streams dictionary is always initialized on program start. If there are no streams, an empty array is returned.
